docs/contributing/contributing-rules.md

```diff
@@ -59,6 +59,8 @@ A well-written rule message includes:
 
 For an example of a good rule message, see: [this rule for Django's mark_safe](https://semgrep.dev/r?q=python.django.security.audit.avoid-mark-safe.avoid-mark-safe).
 
+Use the YAML multiline string operator `>-` when a rule message spans multiple lines. This presents the best-looking rule message in the command-line.
```

Comment:
Do we have an example of this? I checked the above rule (Django mark_safe) but it doesn't have `>-`, just `-`

Reply:
I sometimes see `"`, but nothing else. I was asking Grayson about this previously and he suggested that this is still the case. Please, @minusworld do you know any case where we should still include`>-` for more lines in rules messages? If so, can you please link an example? I tried to search in our rules repository and it seems that it is no longer the case. (For example: https://semgrep.dev/r?q=doctrine-orm-dangerous-query.yaml)

Reply:
The important thing is that there are _no newlines_ in the `message:` key. There are a few ways to achieve this, but the clearest way, in my opinion, is to use `>-`. 

Any way you feel is best to communicate "no newlines in the message key", I'm cool with that. :)

Reply:
The registry backend may reformat rules to be something else, so this may not even be a relevant point anymore. You'll have to ask someone on the App (Onboarding, i think) team if they reformat rules to not have newlines in the backend.

Reply:
Thank you all for help! @efortuna please see below:

## Tested

I tried to test this with a bit weird looking rule:
```yaml
rules:
- id: autofix-markdown-stylistics-examples
  pattern: e.g.
  message: >- 
    This is one line
    of text. Also, when
    we say that "quotes
    are only quotes"
    they are just quotes.
    Semgrep found a stylistic match. The `e.g.` does not conform to Google developer style guide. Non-native English speakers may not understand this abbreviation.
  languages:
    - generic
  severity: WARNING
  fix: for example
  metadata:
    category: best-practice
    technology: markdown
    references: "See https://developers.google.com/style/word-list?hl=en and search for e.g."
```

The output in terminal seems quite nice:
<img width="877" alt="Screenshot 2022-06-24 at 15 01 23" src="https://user-images.githubusercontent.com/26669307/175541549-34aa339c-8580-4703-b9c9-b1814240efa1.png">

## Suggested text

So I suggest to write this as follows:

```markdown
Use the YAML multiline string operator `>-`. Write the rule message as just one long line or multiple lines, they are interpreted as one line. This presents the best-looking rule message on the command-line.

```yaml
message: >-
  This is one line
  of text. Also, when
  we say that "quotes
  are only quotes".
  they are just quotes. 
```

The command-line output will be similar to the following:

```
This is one line of text. Also, when we say that "quotes are only quotes" they are just quotes.
```
```

Reply:
LGTM.

Reply:
Just to move the discussion back here @efortuna - is this better? :-) I agree with your comment [below](https://github.com/returntocorp/semgrep-docs/pull/551#discussion_r906313458), it was too long:

  Use the YAML multiline string operator `>-`. This presents the best-looking rule message on the command-line.

  ```yaml
  message: >-
    This is one line
    of text. Also, when
    we say that "quotes
    are only quotes".
    they are just quotes. 
  ```
  
  The command-line output will be similar to the following:
  
  ```
  This is one line of text. Also, when we say that "quotes are only quotes" they are just quotes.
  ```

Reply:
True, though I will remind you that document you linked is documentation for YAML syntax, while we are not. What about: 

> Use the YAML multiline string operator `>-` when rule messages span multiple lines. This presents the best-looking rule message on the command-line without having to worry about line wrapping or escaping the quote or backslash characters.

Reply:
@efortuna this is awesome! Thank you Emily... I am deleting the code example and using your suggestion:

```markdown
Use the YAML multiline string operator `>-` when rule messages span multiple lines. This presents the best-looking rule message on the command-line without having to worry about line wrapping or escaping the quote or backslash characters.
```